src/libecalc/core/models/compressor/train/simplified_train.py

```diff
@@ -517,12 +512,24 @@ def calculate_maximum_rate_given_outlet_pressure_single_calculation_point(
                     extrapolate_heads_below_minimum=False,
                 )
             )
-            polytropic_efficiency = float(
-                compressor_chart.efficiency_as_function_of_rate_and_head(
+
+            try:
+                efficiency_array: NDArray[np.float64] = compressor_chart.efficiency_as_function_of_rate_and_head(
                     rates=np.atleast_1d(maximum_actual_volume_rate),
                     heads=np.atleast_1d(polytropic_head),
                 )
-            )
+
+                if efficiency_array.size == 0:
+                    raise ValueError("Efficiency array is empty, cannot proceed with calculation.")
+
+                polytropic_efficiency = efficiency_array[0]
+
+            except IndexError as e:
+                raise IndexError("Failed to extract efficiency: array index out of range.") from e
+
+            except Exception as e:
+                raise RuntimeError("An unexpected error occurred during efficiency extraction.") from e
```

Comment:
This isn't necessary IMO. Just let it fail on index lookup instead. If this can actually happen we should either make sure it doesn't or provide some guidance to the user by raising an EcalcError with info.

Reply:
I simplified error handling. Now its basically like before except indexing the array element in a try-catch and raising IndexError if it doesnt work, but it should work since all relevant values further up are floats. Take a look and see if its ok.